src/wallet/wallet_rpc_server.cpp

```diff
@@ -786,7 +786,14 @@ namespace tools
 
     try
     {
-      uint64_t mixin = m_wallet->adjust_mixin(req.mixin);
+      uint64_t mixin;
+      if(req.ring_size != 0)
+      {
+        mixin = m_wallet->adjust_mixin(req.ring_size - 1);
+      }
+      else{
+        mixin = m_wallet->adjust_mixin(req.mixin);
+      }
```

Comment:
Sorry for being a bit  punctilious, but please keep the same brace style at least in a single block. Also below. Monero usually has the opening brace on a new line.